### PR TITLE
[prometheus] vpa.yaml - bump apiVersion(v1beta1 to v1)

### PR DESCRIPTION
**Pull request by arukiidou**
### What this PR does / why we need it

- vpa.yaml - bump apiVersion(v1beta1 to v1)

#### Which issue this PR fixes

- closes https://github.com/prometheus-community/helm-charts/issues/2874

#### Special notes for your reviewer

- refactoring namespace is separated.
  - see: https://github.com/prometheus-community/helm-charts/pull/2883

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/templates/clusterrolebinding.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ metadata:
 subjects:
   - kind: ServiceAccount
     name: {{ template "prometheus.serviceAccountName.server" . }}
-{{ include "prometheus.namespace" . | indent 4 }}
+    namespace: {{ template "prometheus.namespace" . }}
```
Comment (zanhsieh) [inline charts/prometheus/templates/clusterrolebinding.yaml:11]: This is not relevant. Please keep it as-is. Plus, the official document recommend to use `{{ include ... }}` not `{{ template ... }}`.  See:

https://helm.sh/docs/chart_template_guide/named_templates/#the-include-function
https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-include-function

File: charts/prometheus/templates/vpa.yaml [hunk 2]
```diff
@@ -1,11 +1,11 @@
-{{- if .Values.server.verticalAutoscaler.enabled -}}
-apiVersion: autoscaling.k8s.io/v1beta2
+{{- if and (.Capabilities.APIVersions.Has "autoscaling.k8s.io/v1") (.Values.server.verticalAutoscaler.enabled) }}
+apiVersion: autoscaling.k8s.io/v1
```
Comment (zanhsieh) [inline charts/prometheus/templates/vpa.yaml:2]: Just curious if any document (e.g. in README.md) relates to upgrade (e.g. which versions K8S drop support).

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/prometheus/templates/clusterrolebinding.yaml (hunk 1 above)
Comment (arukiidou) [reply charts/prometheus/templates/clusterrolebinding.yaml:11]: Ok, revert and keep it as-is.

File: charts/prometheus/templates/vpa.yaml (hunk 2 above)
Comment (arukiidou) [reply charts/prometheus/templates/vpa.yaml:2]: I think these are out of our scope,

VPA is disable by default and requires pre-installation of CRD.

https://pkg.go.dev/k8s.io/autoscaler/vertical-pod-autoscaler#section-readme

File: charts/prometheus/templates/clusterrolebinding.yaml (hunk 1 above)
Comment (arukiidou) [reply charts/prometheus/templates/clusterrolebinding.yaml]: Done, but a slightly clumsy solution.

Split the namespace fix if you want, do you?

**Comment by zanhsieh** [comment]
@arukiidou 
I would keep non-related change away. Keep it simple and straightforward, which means not to find good excuses for maintainers not approve your PR.

**Comment by arukiidou** [comment]
> @arukiidou I would keep non-related change away. Keep it simple and straightforward, which means not to find good excuses for maintainers not approve your PR.

ok, namespace refactoring will come later

**Review by zanhsieh — APPROVED** [review]

**Review by monotek — COMMENTED** [review]
Do I miss something or will that break for all users still use the old API?

If so major version change and upgrading docs should have been upgraded.

**Comment by arukiidou** [comment]
@monotek 
Very sorry.
I will work on a compatible fix soon

**Comment by arukiidou** [comment]
@monotek

To prevent 19.3.0 from being used, needs 19.3.1 which is compatible with 19.2.2. Which fix do you want?

1. Make a patch to restore compatibility between 19.2.2 and 19.3.1 (#2896)
   - still a draft, because needs additional testing
2. Revert this, and create 19.3.1 which is exactly the same as 19.2.2

**Comment by monotek** [comment]
19.3.1 should add a fix that both api versions are supported, by using "apabilities.APIVersions.Has" function.

**Comment by arukiidou** [comment]
> 19.3.1 should add a fix that both api versions are supported, by using "apabilities.APIVersions.Has" function.

#2896 is both supported v1/v1beta2.

```
{{- if .Values.server.verticalAutoscaler.enabled -}}
{{- if .Capabilities.APIVersions.Has "autoscaling.k8s.io/v1/VerticalPodAutoscaler" }}
apiVersion: autoscaling.k8s.io/v1
{{- else }}
apiVersion: autoscaling.k8s.io/v1beta2
{{- end }}
```

**Pull request merged**